api/types/app.go

```diff
@@ -301,14 +301,16 @@ func (a *AppV3) CheckAndSetDefaults() error {
 	return nil
 }
 
-// DeduplicateApps deduplicates apps by name.
+// DeduplicateApps deduplicates apps by combination of app name and public address.
+// Apps can have the same name but also could have different addresses.
 func DeduplicateApps(apps []Application) (result []Application) {
```

Comment:
Are we ok with `tsh` now getting duplicated app names as a result of this? And what about the other `DeduplicateFoo` functions?